Traceback (most recent call last):
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/Users/yewang/.pyenv/versions/3.9.15/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from sklearn.decomposition import PCA

# 生成示例数据
np.random.seed(0)
X = np.random.rand(100, 5)

# 拟合PCA模型
pca = PCA(n_components=2)
X_pca = pca.fit_transform(X)

# 绘制PCA降维结果
plt.scatter(X_pca[:, 0], X_pca[:, 1])
plt.title('PCA 降维示例')
plt.xlabel('主成分 1')
plt.ylabel('主成分 2')
plt.show()
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [0;32mIn[1], line 1[0m
[0;32m----> 1[0m [38;5;28;01mfrom[39;00m [38;5;21;01msklearn[39;00m[38;5;21;01m.[39;00m[38;5;21;01mdecomposition[39;00m [38;5;28;01mimport[39;00m PCA
[1;32m      3[0m [38;5;66;03m# 生成示例数据[39;00m
[1;32m      4[0m np[38;5;241m.[39mrandom[38;5;241m.[39mseed([38;5;241m0[39m)

[0;31mModuleNotFoundError[0m: No module named 'sklearn'

